server/src/main/java/org/opensearch/cluster/service/MasterThrottlingStats.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.cluster.service;
+
+import org.opensearch.common.metrics.CounterMetric;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * Contains stats of Master Task Throttling.
+ * It stores the total cumulative count of throttled tasks per task type.
+ */
+public class MasterThrottlingStats implements MasterTaskThrottlerListener {
+
+    private Map<String, CounterMetric> throttledTasksCount = new ConcurrentHashMap<>();
+
+    private void incrementThrottlingCount(String type, final int permits) {
+        if (!throttledTasksCount.containsKey(type)) {
+            throttledTasksCount.computeIfAbsent(type, k -> new CounterMetric());
+        }
+        throttledTasksCount.get(type).inc(permits);
+    }
+
+    public long getThrottlingCount(String type) {
```

Comment:
we could use an API to get  throttled metrics along with the task type. It would be really helpful to tune this up as an end user . However, we can do it once the stats API is present as well .